Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -299,11 +299,27 @@ function Wait-ServiceFabricApplicationUpgradeAction
     }
 
     $upgradeStatusValidator = { param($upgradeStatus) return !($upgradeStatus.UpgradeState -eq "RollingBackCompleted" -or $upgradeStatus.UpgradeState -eq "RollingForwardCompleted") }
-    $upgradeStatus = Invoke-ActionWithRetries -Action $upgradeStatusFetcher `
-        -ResultRetryEvaluator $upgradeStatusValidator `
-        -MaxTries 2147483647 `
-        -RetryIntervalInSeconds 5 `
-        -RetryableExceptions @("System.Fabric.FabricTransientException", "System.TimeoutException")
+
+    try
+    {
+        $upgradeStatus = Invoke-ActionWithRetries -Action $upgradeStatusFetcher `
+            -ResultRetryEvaluator $upgradeStatusValidator `
+            -MaxTries 2147483647 `
+            -RetryIntervalInSeconds 5 `
+            -RetryableExceptions @("System.Fabric.FabricTransientException", "System.TimeoutException")
+
+        if($upgradeStatus.UpgradeState -eq "RollingBackCompleted")
```

Comment:
This condition is already checked in Line 366 in Publish-UPgradedServiceFbaricApplicaiton.ps1. Print status there.

Reply:
Motivation behind having similar IF condition at 2 places is that before checking this condition and after fetching upgrade status we are performing couple of operations like Get-ServiceFabricApplicationTypeAction & Unregister-ServiceFabricApplicationTypeAction which can cause their own failure in exceptional cases due to which system will not be able to print cluster/application health.
Also, these operations have their corresponding logs associated and I'd prefer to print Cluster and Application health right fetching the upgrade status.
Likewise there can be multiple such operations can be introduced in future in between, so it'd always be better to print the same right after fetching the status.
Let me know if you have any concern apart from having same IF condition at 2 places because this looks fine to me?